### PR TITLE
Display non fungible data fields

### PR DESCRIPTION
**Pull request by GhenadieVP**
Display NFT Data fields.

# What has been done
- Properly display different kinds of NFT data based on SBOR value.

# Demo

https://github.com/radixdlt/babylon-wallet-ios/assets/118184705/370a7f4c-b28c-4d3e-a0bb-e2ede5b89082

### CODE REVIEW

File: RadixWallet.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -230,7 +230,6 @@
       "kind" : "remoteSourceControl",
       "location" : "https://github.com/radixdlt/swift-engine-toolkit",
       "state" : {
-        "branch" : "add_final_classes",
         "revision" : "7a26ed178eaf0cb842276e1b6aa8d7a4b614713c"
```
Comment (GhenadieVP) [inline RadixWallet.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:233]: Pegged to a specific commit until the RET update PR is merged.

File: RadixWallet/Core/DesignSystem/ExpandableTextView.swift [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+			.textStyle(textStyle)
+			.foregroundColor(foregroundColor)
+			.frame(maxWidth: .infinity, alignment: .leading)
+			.lineLimit(isExpanded ? .none : .some(lineLimit))
+			.onSizeChanged { size in
+				truncatedSize = size
+				isTruncated = truncatedSize != intrinsicSize
+			}
+			.background(
+				Text(.init(text)) // Background text used to determin if the text needs to be truncated or not
```
Comment (CyonAlexRDX) [inline RadixWallet/Core/DesignSystem/ExpandableTextView.swift:39]: determine

File: RadixWallet/Core/DesignSystem/ExpandableTextView.swift [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	@State private var isTruncated: Bool = false
+	@State private var intrinsicSize: CGSize = .zero
+	@State private var truncatedSize: CGSize = .zero
+	@State private var moreTextSize: CGSize = .zero
+
+	private var textStyle: TextStyle = .body1Regular
+	private var foregroundColor: Color = .app.gray1
+	private var textForDisplay: String {
+		if isExpanded {
+			text + "\n" // Add aditional new line to make sure that "Show Less" does not overlay some actual text
```
Comment (CyonAlexRDX) [inline RadixWallet/Core/DesignSystem/ExpandableTextView.swift:17]: additional

File: RadixWallet/Core/SharedModels/LedgerIdentifiable.swift [hunk 4]
```diff
@@ -94,3 +94,22 @@ extension LedgerIdentifiable {
...
+		case _ where ResourceEntityType.addressSpace.contains(address.decodedKind):
+			self = .resource(.init(address: address.address, decodedKind: address.decodedKind))
+		case _ where PackageEntityType.addressSpace.contains(address.decodedKind):
+			self = .package(.init(address: address.address, decodedKind: address.decodedKind))
+		case _ where ValidatorEntityType.addressSpace.contains(address.decodedKind):
+			self = .validator(.init(address: address.address, decodedKind: address.decodedKind))
+		case _ where ComponentEntityType.addressSpace.contains(address.decodedKind):
+			self = .component(.init(address: address.address, decodedKind: address.decodedKind))
+		default:
+			return nil
```
Comment (CyonAlexRDX) [inline RadixWallet/Core/SharedModels/LedgerIdentifiable.swift:112]: can we actually use the switch?  and remove the default?

File: RadixWallet/Features/AssetsFeature/Components/NonFungibleAssetList/Components/Details/NonFungibleTokenDetails+View.swift [hunk 5]
```diff
@@ -86,9 +105,243 @@ extension NonFungibleTokenDetails {
...
+				}
+
+			case let .decimal(value):
+				KeyValueView(key: field.name, value: value.formatted())
+
+			case let .enum(variant):
+				KeyValueView(key: field.name, value: variant)
+
+			case let .id(id):
+				KeyValueView(key: field.name, value: try! id.toString())
```
Comment (CyonAlexRDX) [inline RadixWallet/Features/AssetsFeature/Components/NonFungibleAssetList/Components/Details/NonFungibleTokenDetails+View.swift:159]: can we avoid force `try!` ?

Comment (CyonAlexRDX) [reply RadixWallet/Features/AssetsFeature/Components/NonFungibleAssetList/Components/Details/NonFungibleTokenDetails+View.swift:159]: `if let idString = try? id.toString() {
    KeyValueView(...)
} else {
   // what goes here?
}

File: RadixWallet/Core/SharedModels/LedgerIdentifiable.swift (hunk 4 above)
Comment (GhenadieVP) [reply RadixWallet/Core/SharedModels/LedgerIdentifiable.swift:112]: I don't think so, there are an infinite number of possibilities with this, it is just a nice form for if-else, at least I do prefer it.

File: RadixWallet/Features/AssetsFeature/Components/NonFungibleAssetList/Components/Details/NonFungibleTokenDetails+View.swift (hunk 5 above)
Comment (GhenadieVP) [reply RadixWallet/Features/AssetsFeature/Components/NonFungibleAssetList/Components/Details/NonFungibleTokenDetails+View.swift:159]: sure, though it is assured to not fail. Can fallback to an empty view.

**Review by CyonAlexRDX — APPROVED** [review]
Lgtm

**Pull request merged**